app/acknowledgement/update-acknowledgement.js

```diff
@@ -0,0 +1,33 @@
+const acknowledgementSchema = require('../data/schemas/acknowledgement')
+
+const config = require('../config')
+
+const { sendProcessingAckErrorEvent } = require('../event')
+const acknowledgePaymentRequest = require('./acknowledge-payment-request')
+const getPaymentRequest = require('./get-payment-request')
+const processInvalid = require('./process-invalid')
+
+const updateAcknowledgement = async (acknowledgement) => {
+  const schemaResult = acknowledgementSchema.required().validate(acknowledgement, {
+    abortEarly: false,
+    allowUnknown: false
```

Comment:
I don't think we need to introduce this constraint on the acknowledgement.  As long as we get the minimum we need in the message I don't think it matters if there are additional fields.

For example, we may add other subscribers to acknowledgements beyond this service and they may introduce additional fields.  I wouldn't want to have to make changes here.